carbon-design-system/carbon-addons-iot-react
# feat(card): enable clickable tooltips with external content to title

Closes #3383

**Summary**
Added a new prop called `titleTextTooltip` that accepts an element to be shown as a tooltip when the title is clicked. If the title is overflowing the full title is prepended to the content of the tooltip. Works for all the cards except the TableCard since the TableCard doesn't use the card title.

See discussion in #3383

**Change List (commits, features, bugs, etc)**
- Added new prop `titleTextTooltip` that accepts an element to be shown as a tooltip when the title is clicked
- Added warnings and checks to make this prop mutually exclusive with exiting `tooltip` prop that uses the info icon. Also checks and warns if combined with hasTitleWrap.
- Wrote unit tests for new and existing functionality
- Refactored CardTitle into a new component and separate scss file
- Updated Card story with knobs for `titleTextTooltip`, `tooltip` and `hasTitleWrap`
- Updated DashboardGrid stories with knob for `titleTextTooltip`

**Acceptance Test (how to verify the PR)**
1. OPTIONAL. Pull and run locally, go to story ?path=/story/1-watson-iot-card-card--with-ellipsed-title-tooltip-external-tooltip
2. Enable knob `titleTextTooltip` 
3. Make sure there is a console warning about titleTextTooltip and tooltip cannot be combined

1. Go to story https://deploy-preview-3402--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-card-card--with-ellipsed-title-tooltip-external-tooltip
2. Enable knob `titleTextTooltip` 
4. Verify that there is dotted line under the title similar to as if the DefinitionTooltip would have been used
5. Click the title and make sure the tooltip contains the full title and the external content
6. Click the button in the tooltip and verify that the story action is called 
7. change title knob to contain something short and verify that the title text is removed from the tooltip content

8. Go to  https://deploy-preview-3402--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-dashboard--dashboard-all-card-sizes 
9. Enable knob "Demo titleTextTooltip"
10. Verify that the tooltip doesn't open outside the screen for the different cards

11. Go to https://deploy-preview-3402--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-dashboard--dashboard-all-cards-as-resizable
12. Enable knob "Demo titleTextTooltip"
13. Verify that the tooltip is not covered by any of the cards content, eg. charts


**Regression Test (how to make sure this PR doesn't break old functionality)**
- Check all the card stories. The card still works as previous. Check especially the old (info icon) tooltip and the subtitle tooltip and make sure they haven't changed



**Things to look for during review**
